classes/Services/AddThisScriptManager.php

```diff
@@ -0,0 +1,245 @@
+<?php
+/**
+ * @file
+ * Class definition of a script manager.
+ *
+ * This class will be used on different places. The result of the attachJsToElement()
+ * should be the same in every situation within one request and throughout the
+ * loading of the site.
+ *
+ * When manipulating the configuration do this very early in the request. This
+ * could be hook_init() for example. Any other method should be before hook_page_build().
+ * The implementation of addthis_page_build() is the first known instance where
+ * this class might get used based on the configuration.
+ */
+
+class AddThisScriptManager {
+
+  private $addthis = NULL;
+  private $async = NULL;
+  private $domready = NULL;
+
+  /**
+   * Construct method.
+   */
+  private function __construct() {
+    $this->addthis = AddThis::getInstance();
+
+    $this->async = $this->addthis->getWidgetJsAsync();
+    $this->domready = $this->addthis->getWidgetJsDomReady();
+  }
+
+  /**
+   * Return a single instance of the AddThisScriptManager.
+   *
+   * @return AddThisScriptManager
+   */
+  public static function getInstance() {
+    static $manager;
+
+    if (!isset($manager)) {
+      $manager = new AddThisScriptManager();
+    }
+    return $manager;
+  }
+
+  /**
+   * Get the current widget js url.
+   * 
+   * @return string
+   *   A url reference to the widget js.
+   */
+  public function getWidgetJsUrl() {
+    return check_url(variable_get(AddThis::WIDGET_JS_URL_KEY, AddThis::DEFAULT_WIDGET_JS_URL));
+  }
+
+  /**
+   * Return if we are on https connection.
+   * 
+   * @return bool
+   *   TRUE if the current request is on https. 
+   */
+  public function isHttps() {
+    $is_https = isset($_SERVER['HTTPS']) && strtolower($_SERVER['HTTPS']) == 'on';
+
+    return $is_https;
+  }
+
+  /**
+   * Change the schema from http to https if we are on https.
+   * 
+   * @param  string $url
+   *   A full url.
+   * 
+   * @return string
+   *   The changed url.
+   */
+  public function correctSchemaIfHttps($url) {
+    if (is_string($url) && $this->isHttps()) {
+      return str_replace('http://', 'https://', $url);
+    } 
+    else {
+      return $url;
+    }
+    throw new InvalidArgumentException('The argument was not a string value');
+  }
+
+  /**
+   * Attach the widget js to the element.
+   * 
+   * @todo Change the scope of the addthis.js.
+   *   See if we can get the scope of the addthis.js into the header
+   *   just below the settings so that the settings can be used in the loaded
+   *   addthis.js of our module.
+   *
+   * @param array $element
+   *   The element to attach the JavaScript to.
+   */
+  public function attachJsToElement(&$element) {
+    $attachments = &drupal_static(__FUNCTION__);
```

Comment:
Ah okay, might be frustrating in some cases because some other code already set the settings. Maybe add some alter here at the bottom?